backend/module/chaos/serverexperimentation/rtds/rtds_cache.go

```diff
@@ -104,61 +115,73 @@ func PeriodicallyRefreshCache(s *Server) {
 	go func() {
 		for range ticker.C {
 			s.logger.Info("Refreshing RTDS cache")
-			refreshCache(s.ctx, s.storer, &cacheWrapperV2{s.snapshotCacheV2}, s.rtdsLayerName, s.logger)
-			refreshCache(s.ctx, s.storer, &cacheWrapperV3{s.snapshotCacheV3}, s.rtdsLayerName, s.logger)
+			refreshCache(s.ctx, s.storer, &cacheWrapperV2{s.snapshotCacheV2}, s.rtdsLayerName, s.ingressPrefix, s.egressPrefix, s.logger)
+			refreshCache(s.ctx, s.storer, &cacheWrapperV3{s.snapshotCacheV3}, s.rtdsLayerName, s.ingressPrefix, s.egressPrefix, s.logger)
 		}
 	}()
 }
 
 func refreshCache(ctx context.Context, storer experimentstore.Storer, snapshotCache cacheWrapper, rtdsLayerName string,
-	logger *zap.SugaredLogger) {
-	allExperiments, err := storer.GetExperiments(ctx, "type.googleapis.com/clutch.chaos.serverexperimentation.v1.TestConfig", experimentation.GetExperimentsRequest_STATUS_RUNNING)
+	ingressPrefix string, egressPrefix string, logger *zap.SugaredLogger) {
+	allRunningExperiments, err := storer.GetExperiments(ctx, "type.googleapis.com/clutch.chaos.serverexperimentation.v1.TestConfig", experimentation.GetExperimentsRequest_STATUS_RUNNING)
 	if err != nil {
 		logger.Errorw("Failed to get data from experiments store", "error", err)
 
 		// If failed to get data from DB, stop all ongoing faults.
-		allExperiments = []*experimentation.Experiment{}
+		allRunningExperiments = []*experimentation.Experiment{}
 	}
 
-	// Group faults by upstream cluster
-	upstreamClusterFaultMap := make(map[string][]*experimentation.Experiment)
-	for _, experiment := range allExperiments {
+	clusterFaultMap := make(map[string][]*experimentation.Experiment)
+	for _, experiment := range allRunningExperiments {
 		testConfig := &serverexperimentation.TestConfig{}
 		if !isFaultTest(experiment, testConfig) {
 			continue
 		}
-		clusterPair := testConfig.GetClusterPair()
-		upstreamClusterFaultMap[clusterPair.UpstreamCluster] =
-			append(upstreamClusterFaultMap[clusterPair.UpstreamCluster], experiment)
+
+		upstreamCluster := testConfig.GetClusterPair().GetUpstreamCluster()
+		downstreamCluster := testConfig.GetClusterPair().GetDownstreamCluster()
+		faultInjectionCluster := testConfig.GetClusterPair().GetFaultInjectionCluster()
+
+		switch faultInjectionCluster {
+		case serverexperimentation.FaultInjectionCluster_FAULTINJECTIONCLUSTER_UPSTREAM:
+			clusterFaultMap[upstreamCluster] = append(clusterFaultMap[upstreamCluster], experiment)
+
+		case serverexperimentation.FaultInjectionCluster_FAULTINJECTIONCLUSTER_DOWNSTREAM:
+			clusterFaultMap[downstreamCluster] = append(clusterFaultMap[downstreamCluster], experiment)
+
+		default:
+			logger.Errorw("Invalid fault injection cluster found", "upstream", upstreamCluster, "downstream", downstreamCluster, "faultInjectionCluster", faultInjectionCluster)
+			panic("Unknown fault injection cluster")
```

Comment:
can we not panic here? maybe we could just clean-up (remove) all runtime override instead? (if it's easy to do)

Reply:
or we can just `continue` here - I think it should be safe to `continue` here, The log of an error will trigger alarm and we will be able to investigate.